wordvecspace/wvspace.py

```diff
@@ -231,33 +228,301 @@ def _nearest_sorting(self, d, k):
+        ... combination=True, combination_method='set_union', weights=[0.1, 0.9])
+        array([[13942,   503, 28413, 27280, 26180]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500,
+        ... combination=True, combination_method='set_union', weights=[0.9, -0.9])
+        array([[ 2822,  2765, 12515, 66318,  8591]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500, combination=True,
+        ... combination_method='set_union', weights=[0.1, 0.9], include_distances=True)[1]
+        array([[0.0321, 0.0321, 0.0322, 0.0322, 0.0322]], dtype=float32)
+	>>> wv.get_nearest(vecs, k=5, combination=True, combination_method='distance')
+	array([[  523,  5969, 24149,  9772,   486]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True,
+        ... combination_method='distance', weights=[0.5, 0.3])
+        array([[  523,  5969,   486, 24149,  9772]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True,
+        ... combination_method='distance', weights=[0.5, 0.3], include_distances=True)[1]
+        array([[0.2425, 0.2435, 0.2435, 0.2439, 0.2441]], dtype=float32)
+	>>> wv.get_nearest(vecs, k=5, combination=True, combination_method='vector')
+	array([[  523,  5969, 24149,  9772,   486]], dtype=uint32)
+	>>> wv.get_nearest(vecs, k=5, combination=True,
+	... combination_method='vector', weights=[0.8, 0.2])
+	array([[  486,   509,   523, 24149,  3389]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True,
+        ... combination_method='vector', weights=[0.5, 0.3])
+        array([[  523,  5969,   486, 24149,  9772]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True,
+        ... combination_method='vector', weights=[0.5, 0.3], include_distances=True)[1]
+        array([[0.0006, 0.0023, 0.0023, 0.003 , 0.0034]], dtype=float32)
+        """
+
+        if not combination:
+            return self._get_brute(v_w_i, k, metric,
+                                   combination, include_distances)
+
+        if combination and combination_method == 'set_intersect':
+            return self._get_set_intersect(v_w_i, k, examine_k, metric,
+                                           weights, include_distances)
+
+        if combination and combination_method == 'set_union':
+            return self._get_set_union(v_w_i, k, examine_k, metric,
+                                       weights, include_distances)
+
+        if combination and combination_method == 'distance':
+            return self._get_weighted_brute(v_w_i, k, metric,
+                                            weights, include_distances)
+
+        if combination and combination_method == 'vector':
+            return self._get_resultant_vector_nearest(v_w_i, k, metric,
+                                                      weights, include_distances)
+
+    def _get_set_intersect(self, v_w_i, k, examine_k, metric, weights, include_distances):
```

Comment:
I think this function is written twice, please make sure that we don't duplicate the code.